### PR TITLE
Fix links in `troubleshooting.md`

### PR DESCRIPTION
**Pull request by hornc**
Tiny typo/link fix to Troubleshooting documentation. No functionality changed, just documentation.
## Description

Previous link 404s with https://github.com/github/linguist/blob/master/docs/CONTRIBUTING.md

This corrects it to the intended https://github.com/github/linguist/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by Alhadis — CHANGES_REQUESTED** [review]
Nice catch!

While we're here, we might as well fix the other broken links I've just noticed (under _“My repository is detected as the wrong language”_):

1.	> 3. If the files are misclassified, search for [open issues][issues]  
	>    …
	> 4. If there are no reported issues of this misclassification, [open an issue][new-issue]

	The missing link definitions are vestiges of [#5202](https://github.com/github/linguist/pull/5202). Add:

	~~~diff
	+[issues]: https://github.com/github/linguist/issues
	+[new-issue]: https://github.com/github/linguist/issues/new
	 [search-limits]: https://docs.github.com/github/searching-for-information-on-github/searching-code#considerations-for-code-search
	~~~

2.	> Keep in mind that the repository language stats are only [updated when you push changes](#how-linguist-works-on-githubcom),
	
	That link should point to [`/docs/how-linguist-works.md`](https://github.com/github/linguist/blob/master/docs/how-linguist-works.md)

**Review by Alhadis — APPROVED** [review]

**Comment by hornc** [comment]
Thanks @Alhadis , I noticed the `[issues]` and `[new-issues]` after I raised the PR but wasn't 100% sure if they were supposed to be active links. I looked for GitHub markdown shortcuts to the actions; full URLs are a sensible answer :)

I have made those changes, and one other: changing 'above' to 'here' on the single file link back to README.md.

Thanks for the context with the original documentation split PR !

**Comment by Alhadis** [comment]
Great work! We'll merge this once @lildude has given his 👍 of approval.

**Review by lildude — APPROVED** [review]

**Pull request merged**
